components/gantry/singleaxis/singleaxis.go

```diff
@@ -196,9 +210,80 @@ func (g *singleAxis) Reconfigure(ctx context.Context, deps resource.Dependencies
 		wg.Wait()
 	}
 
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
+	g.checkHit(ctx)
```

Comment:
Just noticed something - if there are no limit switches after reconfiguring - do you want to start the check at all?